levelpacker/levelpacker.py

```diff
@@ -55,32 +62,41 @@ def handle_objects_layer(g: minidom.Node, objs: GameObjects, builder: flatbuffer
         FlatBuffGenerated.Object.ObjectAddPos(builder, pos)
         FlatBuffGenerated.Object.ObjectAddRotation(builder, rot)
         FlatBuffGenerated.Object.ObjectAddGid(builder, gid)
+        size = FlatBuffGenerated.Vec2.CreateVec2(builder, width, height)
+        FlatBuffGenerated.Object.ObjectAddSize(builder, size)
         FlatBuffGenerated.Object.ObjectAddHspotname(builder, hspotNameFb)
         objs.objects.append(FlatBuffGenerated.Object.ObjectEnd(builder))
 
 def handle_decoration_layer(g: minidom.Node, objs: GameObjects, builder: flatbuffers.Builder):
     for o in g.getElementsByTagName('object'):
         x, y, rot = get_pos(o)
+        width, height = get_dim(o)
         gid = int(o.getAttribute('gid'))
 
         FlatBuffGenerated.Decoration.DecorationStart(builder)
         pos = FlatBuffGenerated.Vec2.CreateVec2(builder, x * GLOBAL_SCALE, y * GLOBAL_SCALE)
         FlatBuffGenerated.Decoration.DecorationAddPos(builder, pos)
         FlatBuffGenerated.Decoration.DecorationAddRotation(builder, rot)
+        size = FlatBuffGenerated.Vec2.CreateVec2(builder, width, height)
+        FlatBuffGenerated.Decoration.DecorationAddSize(builder, size)
         FlatBuffGenerated.Decoration.DecorationAddGid(builder, gid)
         objs.decoration.append(FlatBuffGenerated.Decoration.DecorationEnd(builder))
 
 def handle_collision_layer(g: minidom.Node, objs: GameObjects, builder: flatbuffers.Builder):
     for o in g.getElementsByTagName('object'):
         x, y, rotation = get_pos(o)
-        radius = 0
         poly = 0
-
-        if o.getElementsByTagName('ellipse'):
-            assertCircleness(o)
-            radius = float(o.getAttribute('width')) / 2.0
-            x += math.cos(math.radians(rotation) + math.radians(45)) * radius * math.sqrt(2) # trygonometria 100
-            y += math.sin(math.radians(rotation) + math.radians(45)) * radius * math.sqrt(2)
+        width = 0
+        height = 0
+        isCircle = False
+
+        if not o.getElementsByTagName('polygon'):
+            width, height = get_dim(o)
+            x += (math.cos(math.radians(rotation)) * width/2.0  - math.sin(math.radians(rotation)) * height/2.0)
+            y += (math.cos(math.radians(rotation)) * height/2.0 + math.sin(math.radians(rotation)) * width/2.0)
+            if o.getElementsByTagName('ellipse'):
+                assertCircleness(o)
+                isCircle = True
```

Comment:
i don't think any of this is related to tiles. i'll merge this in this pr but in the future please add functionalities in relevant prs